components/user-manager/org.wso2.carbon.user.mgt.ui/src/main/resources/web/user/add-finish.jsp

```diff
@@ -55,9 +55,10 @@
             claims = new ClaimValue[]{emailClaim};
             userPassword = null;
         }
-        userBean.addUserRoles((Map<String,Boolean>)session.getAttribute("checkedRolesMap"));
-        client.addUser(Util.decodeHTMLCharacters(username), userPassword, userBean.getUserRoles(), claims, null);
+        String[] userRoles = userBean.getUserRoles();
+        userBean.removeUserRoles();
```

Comment:
clear this session in 'finally', remove the user bean from session and forward to user-mgt.jsp, remove all session attributes in finally bock